### PR TITLE
8341057: Add 2 SSL.com TLS roots

### PR DESCRIPTION
**Pull request by gnu-andrew**
Hi all,

This pull request contains a backport of commit [217b9fdf](https://github.com/openjdk/jdk11u/commit/217b9fdf056f4887230f60fdf60b036861699cd8) from the [openjdk/jdk11u](https://git.openjdk.org/jdk11u) repository.

The commit being backported was authored by Goetz Lindenmaier on 1st of October, 2024 and was reviewed by Matthias Baesken.

The patch applied as is to 8u once path shuffled. All enabled `security/infra/java/security/cert/CertPathValidator/certification` tests pass, including the new `ssltlsrootecc2022` and `ssltlsrootrsa2022`. These new certificates [will be used as the issuing CA by Entrust after the 12th of November](https://bugs.openjdk.org/browse/JDK-8341087).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8341057](https://bugs.openjdk.org/browse/JDK-8341057) needs maintainer approval

### Issue
 * [JDK-8341057](https://bugs.openjdk.org/browse/JDK-8341057): Add 2 SSL.com TLS roots (**Enhancement** - P2 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/589/head:pull/589` \
`$ git checkout pull/589`

Update a local copy of the PR: \
`$ git checkout pull/589` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/589/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 589`

View PR using the GUI difftool: \
`$ git pr show -t 589`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/589.diff">https://git.openjdk.org/jdk8u-dev/pull/589.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/589#issuecomment-2387478672)

### CODE REVIEW

File: jdk/test/security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java [hunk 1]
```diff
@@ -484,6 +484,28 @@
...
 
+/*
+ * @test id=ssltlsrootecc2022
+ * @bug 8341057
+ * @summary Interoperability tests with SSL TLS 2022 root CAs
+ * @library /test/lib
+ * @build jtreg.SkippedException ValidatePathWithURL CAInterop
+ * @run main/othervm/manual -Djava.security.debug=certpath,ocsp CAInterop ssltlsrootecc2022 DEFAULT
+ * @run main/othervm/manual -Djava.security.debug=certpath,ocsp -Dcom.sun.security.ocsp.useget=false CAInterop ssltlsrootecc2022 DEFAULT
+ * @run main/othervm/manual -Djava.security.debug=certpath CAInterop ssltlsrootecc2022 CRL
```
Comment (jerboaa) [inline jdk/test/security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java:495]: In 8u we usually run those tests (not specifying them as manual). That's why they weren't run in GHA. Please remove `/manual`.

File: jdk/test/security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java [hunk 2]
```diff
@@ -484,6 +484,28 @@
...
+
+/*
+ * @test id=ssltlsrootrsa2022
+ * @bug 8341057
+ * @summary Interoperability tests with SSL TLS 2022 root CAs
+ * @library /test/lib
+ * @build jtreg.SkippedException ValidatePathWithURL CAInterop
+ * @run main/othervm/manual -Djava.security.debug=certpath,ocsp CAInterop ssltlsrootrsa2022 DEFAULT
+ * @run main/othervm/manual -Djava.security.debug=certpath,ocsp -Dcom.sun.security.ocsp.useget=false CAInterop ssltlsrootrsa2022 DEFAULT
+ * @run main/othervm/manual -Djava.security.debug=certpath CAInterop ssltlsrootrsa2022 CRL
```
Comment (jerboaa) [inline jdk/test/security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java:506]: Same here. Remove `/manual`

File: jdk/test/security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java (hunk 1 above)
Comment (gnu-andrew) [reply jdk/test/security/infra/java/security/cert/CertPathValidator/certification/CAInterop.java]: Ah yes, good catch. This is going to be a pain with these backports going forward.

**Comment by gnu-andrew** [comment]
Should be fixed with the latest commit. `git diff dev/master|grep manual` is now empty.

**Review by jerboaa — APPROVED** [review]
LGTM

**Comment by gnu-andrew** [comment]
Thanks Severin.

/approval request New SSL.com TLS roots are necessary as they will be the issuing CA used by Entrust going forward (see JDK-8341087) Patch was not clean but was reviewed by Severin Gehwolf.

**Pull request closed without merging**